trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.get.test.ts

```diff
@@ -0,0 +1,114 @@
+import { createMocks } from 'node-mocks-http';
+import { DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import { aTfmSessionUser } from '../../../../test-helpers';
+import { getReasonForCancelling, GetReasonForCancellingRequest } from './reason-for-cancelling.controller';
+import { PRIMARY_NAVIGATION_KEYS } from '../../../constants';
+import { ReasonForCancellingViewModel } from '../../../types/view-models';
+import api from '../../../api';
+import { canSubmissionTypeBeCancelled } from '../../helpers';
+
+jest.mock('../../../api', () => ({
+  getDeal: jest.fn(),
+  getDealCancellation: jest.fn(),
+}));
+
+jest.mock('../../helpers', () => ({
+  canSubmissionTypeBeCancelled: jest.fn(() => true),
+}));
+
+const dealId = 'dealId';
+const ukefDealId = 'ukefDealId';
+const mockUser = aTfmSessionUser();
+
+describe('getReasonForCancelling', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.mocked(canSubmissionTypeBeCancelled).mockReturnValue(true);
+  });
+
+  it('redirects to not found if the deal does not exist', async () => {
```

Comment:
An assertion for a non-PIM user please.

Reply:
The PIM user's access is prevented in middleware so cant be tested in this unit test.

TFM-UI doesn't have any api-tests currently, which makes this a bit harder to test - I'd like to add some but I think we'll need to get confirmation from Tom that hes happy with that, and to check this wont conflict too painfully with SSO for @AlexBramhill

Reply:
Can we not mock a PIM user in the user session and assert with the expected behaviour?

```typescript
  session: {
        user: mockPimUser,
        userToken: 'a pim user token',
      },
```

Reply:
@abhi-markan Mocking this in this test won't do anything as the PIM validation happens in middleware outside of this test file. 
We have two options: 
1) test this in an api-test (preferred) 
OR 2) test this in an E2E test (breaks consistency)

With option 1, this is what we'd prefer to do, but we currently don't have any api-tests at all in `tfm-ui`. Mark has a ticket up for this and I've noted that we should test the PIM/non-PIM users in deal cancellation there when that ticket is scoped and run by Tom. https://ukef-dtfs.atlassian.net/browse/DTFS2-7363

With option 2, we don't currently do this for e2e tests (as in e.g. for amendments we don't check non-PIM users can't amend deals in an e2e test so this would introduce a new pattern). 

So our preference is 1, and that will be deferred to a separate PR as part of ticket 7363 linked to above when that gets approved. How does that sound?

Reply:
I don't think we need to test a PIM user here at all @abhi-markan as it is handled in middleware which will be called before the controller is hit